bots/merge/src/main/java/org/openjdk/skara/bots/merge/MergeBotFactory.java

```diff
@@ -38,6 +40,96 @@ public String name() {
         return "merge";
     }
 
+    private static MergeBot.Spec.Frequency.Interval toInterval(String s) {
+        switch (s.toLowerCase()) {
+            case "hourly":
+                return MergeBot.Spec.Frequency.Interval.HOURLY;
+            case "daily":
+                return MergeBot.Spec.Frequency.Interval.DAILY;
+            case "weekly":
+                return MergeBot.Spec.Frequency.Interval.WEEKLY;
+            case "monthly":
+                return MergeBot.Spec.Frequency.Interval.MONTHLY;
+            case "yearly":
+                return MergeBot.Spec.Frequency.Interval.YEARLY;
+            default:
+                throw new IllegalArgumentException("Unknown interval: " + s);
+        }
+    }
+
+    private static DayOfWeek toWeekday(String s) {
+        switch (s.toLowerCase()) {
+            case "monday":
+                return DayOfWeek.MONDAY;
+            case "tuesday":
+                return DayOfWeek.TUESDAY;
+            case "wednesday":
+                return DayOfWeek.WEDNESDAY;
+            case "thursday":
+                return DayOfWeek.THURSDAY;
+            case "friday":
+                return DayOfWeek.FRIDAY;
+            case "saturday":
+                return DayOfWeek.SATURDAY;
+            case "sunday":
+                return DayOfWeek.SUNDAY;
+            default:
+                throw new IllegalArgumentException("Unknown weekday: " + s);
+        }
+    }
+
+    private static Month toMonth(String s) {
+        switch (s.toLowerCase()) {
```

Comment:
Would perhaps be a little less verbose to use Month.from with an appropriate formatter, but can keep it as-is if you prefer. :)

Reply:
Yeah, I agree, but since I've already written the verbose version, lets just keep it for now :smiley: